compliance-api/src/compliance_api/__init__.py

```diff
@@ -82,6 +83,21 @@ def set_origin():
             )
             if not is_compliance_in_groups:
                 raise PermissionDeniedError("Access Denied")
+
+            # Get the auth_user_id from token (preferred_username field)
+            auth_user_id = token_info.get("preferred_username")
+            if auth_user_id:
+                # Validate that there's an active staff user for this auth_user_id using cache
+                staff_user = CachedStaffUserService.get_staff_by_auth_guid(auth_user_id)
+                if not staff_user:
+                    raise PermissionDeniedError(
+                        "No valid staff user found for this account"
+                    )
+                # Store staff user info in global context for use in endpoints
+                g.staff_user = staff_user
```

Comment:
i am not too sure of saving whole staff_user in the cache ? will it be session bound?or make something like a dataclass?